c/parallel/test/algorithm_execution.h

```diff
@@ -0,0 +1,174 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cuda.h>
+
+#include <iostream>
+#include <optional>
+#include <string>
+
+#include "test_util.h"
+#include <c2h/catch2_test_helper.h>
+#include <cccl/c/types.h>
+
+template <int device_id_ = 0>
+class BuildInformation
+{
+  int cc_major;
+  int cc_minor;
+  const char* cub_path;
+  const char* thrust_path;
+  const char* libcudacxx_path;
+  const char* ctk_path;
```

Comment:
Nitpick: Storing `std::string_view` is usually a bit icky, but in this case it might be nice

Reply:
Changing the type of class member from `const char*` to `std::string_view` would require getter methods to become somewhat more complicated:

```cpp
  const char* get_cub_path() const
  {
    return cub_path.data();
  }
```

Of course, since constructors of `BuildInformation`  class are private and the class implements singleton design to cache device's compute capability, I think it is an unnecessary complication.